bam_masterdata/metadata/entities.py

```diff
@@ -497,6 +500,50 @@ class ObjectType(BaseEntity):
     properties.
     """
 
+    def __setattr__(self, key, value):
+        if key == "_property_metadata":
+            super().__setattr__(key, value)
+            return
+        if not isinstance(self._property_metadata[key], PropertyTypeAssignment):
+            return super().__setattr__(key, value)
+
+        data_type = self._property_metadata[key].data_type
+        if data_type == "CONTROLLEDVOCABULARY":
+            vocabulary_code = self._property_metadata[key].vocabulary_code
+            if not vocabulary_code:
+                raise ValueError(
+                    f"Property '{key}' of type CONTROLLEDVOCABULARY must have a vocabulary_code defined."
+                )
+
+            # get the class of the vocabulary type
+            for file in listdir_py_modules(DATAMODEL_DIR):
+                if "vocabulary_types.py" in file:
+                    vocab_path = file
+                    break
+
+            module = import_module(vocab_path)
+
+            for name, obj in inspect.getmembers(module, inspect.isclass):
+                if issubclass(obj, VocabularyType) and obj is not VocabularyType:
+                    if hasattr(obj, "defs") and obj.defs.code == vocabulary_code:
+                        vocabulary_class = obj()
+
+            codes = []
```

Comment:
I think this logic to get the `codes` of each term in a VocabularyType is already implemented by accessing the `terms` attribute. 

`terms` work for a VocabularyType in a similar fashion as `properties` do for ObjectType. So you could try:
```python
codes = [term.defs.code if term.defs for term in vocabulary_class.terms]
```